src/jit/codegenlegacy.cpp

```diff
@@ -8973,47 +8960,84 @@ void CodeGen::genCodeForCopyObj(GenTreePtr tree, regMaskTP destReg)
     genCodeForTree_DONE(tree, REG_NA);
 
 #ifdef _TARGET_ARM_
-    if (tree->AsBlkOp()->IsVolatile())
+    if (cpObjOp->IsVolatile())
     {
         // Emit a memory barrier instruction after the CopyBlk
         instGen_MemoryBarrier();
     }
 #endif
+    return;
 }
 
 void CodeGen::genCodeForBlkOp(GenTreePtr tree, regMaskTP destReg)
 {
     genTreeOps oper    = tree->OperGet();
-    GenTreePtr op1     = tree->gtOp.gtOp1;
-    GenTreePtr op2     = tree->gtGetOp2();
+    GenTreePtr dest    = tree->gtOp.gtOp1;
+    GenTreePtr src     = tree->gtGetOp2();
     regMaskTP  needReg = destReg;
```

Comment:
I guess this is codegenlegacy... so maybe not worth it

---
In reply to: [77252985](https://github.com/dotnet/coreclr/pull/7027#discussion_r77252985) [](ancestors = 77252985)